Backports for 1.12.0-rc1

KristofferC opened the pull request.

tecosaur committed:
Much more conservative highlighting for 1.12

Following a few vigorous complaints about the highlighting scheme
introduced by JuliaSyntaxHighlighting, I've pared back the default
behaviour to something much more conservative.

(cherry picked from commit 9a46bf56b6ae12faade2aadba66cae579ea49e22)

IanButterworth committed:
Fix nthreadpools size in JLOptions (#58937)

(cherry picked from commit 2349f431b1d0f97f3f06f8adc5d335b89d5f061c)

visr committed:
Add juliac Artifacts.toml in Makefile (#58936)

(cherry picked from commit 6a9da96740c2c49fa8fc9477e6e1376cc6ebf908)

KristofferC committed:
shield `_artifact_str` function behind a world age barrier (#58957)

We already do this for `require` in Base loading, it probably makes
sense to do this here as well, as invalidating this function easily adds
+1s in load time for a jll. Avoids the big load time penalty from
loading IntelOpenMP_jll in
https://github.com/JuliaLang/julia/issues/57436#issuecomment-3052258775.

Before:

```
julia> @time using ModelingToolkit
  6.546844 seconds (16.09 M allocations: 938.530 MiB, 11.13% gc time, 16.35% compilation time: 12% of which was recompilation)
```

After:

```
julia> @time using ModelingToolkit
  5.637914 seconds (8.26 M allocations: 533.694 MiB, 11.47% gc time, 3.11% compilation time: 17% of which was recompilation)
```

---------

Co-authored-by: KristofferC <kristoffer.carlsson@juliacomputing.com>
Co-authored-by: Cody Tapscott <84105208+topolarity@users.noreply.github.com>
(cherry picked from commit 45e0925b67c35cff01b82d730dd850f31eb042c2)

Taaitaaiger commented:
I'm not sure if it got lost along the way or is WIP, but #58540 still needs to be backported

Taaitaaiger commented:
Ah, my bad. I was looking for `jl_get_global_value` in julia.h, but it had been moved to julia_internal.h in a later commit.

DilumAluthgeBot committed:
🤖 [backports-release-1.12] Bump the NetworkOptions stdlib from c090626 to 532992f (#58883)

Stdlib: NetworkOptions
URL: https://github.com/JuliaLang/NetworkOptions.jl.git
Stdlib branch: master
Julia branch: backports-release-1.12
Old commit: c090626
New commit: 532992f
Julia version: 1.12.0-beta4
NetworkOptions version: 1.3.0(Does not match)
Bump invoked by: @DilumAluthge
Powered by:
[BumpStdlibs.jl](https://github.com/JuliaLang/BumpStdlibs.jl)

Diff:
https://github.com/JuliaLang/NetworkOptions.jl/compare/c090626d3feee6d6a5c476346d22d6147c9c6d2d...532992fcc0f1d02df48374969cbae37e34c01360

```
$ git log --oneline c090626..532992f
532992f Merge pull request #39 from JuliaLang/kc/init
3169a7c Merge pull request #37 from visr/ssl_cert_dir_last
8c93b06 [CI] Update GitHub Actions setup (#38)
65eacbd More explicitly test precedence
9c0a877 Search SSL_CERT_FILE before SSL_CERT_DIR
```

Co-authored-by: DilumAluthge <5619885+DilumAluthge@users.noreply.github.com>
